# EDA Datasets Youtube

## Membaca csv datasets dari Great Britain
Datasets Great Britain terdiri dari :
 - GB Comments : datasets komentar dari tiap video
 - GB Videos : datasets tentang video
 - GB Category : datasets kategori video

In [1]:
#import
import pandas as pd
import json as js
from pandas.io.json import json_normalize

#datasets GB
dfGBcomments = pd.read_csv('GBcomments.csv', sep=',', skiprows=0, error_bad_lines=False)
dfGBvideos = pd.read_csv('GBvideos.csv', sep=',', skiprows=0, error_bad_lines=False)

with open('GB_category_id.json') as file:    
    jsGBcategory = js.load(file)

b'Skipping line 113225: expected 4 fields, saw 5\n'
b'Skipping line 158379: expected 4 fields, saw 7\nSkipping line 241590: expected 4 fields, saw 5\nSkipping line 245637: expected 4 fields, saw 7\n'
b'Skipping line 521402: expected 4 fields, saw 5\n'
b'Skipping line 2398: expected 11 fields, saw 21\nSkipping line 2797: expected 11 fields, saw 21\n'


## GB Comments
GB comments adalah dataset komentar video YouTube di Great Britain yang terdiri dari :
 - id video
 - teks komentar
 - berapa jumlah like dari komentar
 - berapa jumlah jawaban dari komentar
 
Tetapi ada data yang tidak dapat terbaca sehingga harus diskip yaitu :
 - line 113225
 - line 158379
 - line 241590
 - line 245637
 - line 521402
 
Dimensi data ini adalah 718451 baris × 4 kolom

In [2]:
dfGBcomments

video_id                                       comment_text  likes  \
0       jt2OHQh0HoQ  It's more accurate to call it the M+ (1000) be...      0   
1       jt2OHQh0HoQ              To be there with a samsung phone\n😂😂😂      1   
2       jt2OHQh0HoQ  Thank gosh, a place I can watch it without hav...      0   
3       jt2OHQh0HoQ  What happened to the home button on the iPhone...      0   
4       jt2OHQh0HoQ  Power is the disease.  Care is the cure.  Keep...      0   
5       jt2OHQh0HoQ  Keep calm and buy iphone 8 Keep calm and buy i...      0   
6       jt2OHQh0HoQ      i am a big fan of youtube and u !!!!!!!!!!!!!      0   
7       jt2OHQh0HoQ  You will never find Losers who line up and pay...      0   
8       jt2OHQh0HoQ  *APPLE JUST COMMENTED ON MY LAST VIDEO* I'm cr...      0   
9       jt2OHQh0HoQ  I'm only here to see Emma, I love her so much!...      0   
10      jt2OHQh0HoQ           Mom said: Just one more video before bed      0   
11      jt2OHQh0HoQ                           what is the song?? 39:20      0   
12      jt2OHQh0HoQ  I started crying at 49:01 when Emma's EP showe...      0   
13      jt2OHQh0HoQ  3 years late on wireless charging, 3 years lat...      0   
14      jt2OHQh0HoQ                             more one before to bed      0   
15      jt2OHQh0HoQ                subscribe me if you love your mummy      0   
16      jt2OHQh0HoQ  They really just took the samsung s8 and named...      0   
17      jt2OHQh0HoQ  so if someone has a twin sibling both can unlo...      0   
18      jt2OHQh0HoQ             For god sake its better than my ps4pro      0   
19      jt2OHQh0HoQ     7mp cam what kind of standard is this in 2017?      0   
20      jt2OHQh0HoQ         1:22:13 he was waiting for people to 👏clap      1   
21      jt2OHQh0HoQ  sarahah has a leak in its system   revealhater...      0   
22      jt2OHQh0HoQ  If apple sucks and samsungs better like this c...      0   
23      jt2OHQh0HoQ  This video was too long. Halfway through it, I...      0   
24      jt2OHQh0HoQ  You Kicks Steve Jobs out the Company;( and do ...      0   
25      jt2OHQh0HoQ  Love is all you need.... yeah, love and a shin...      0   
26      jt2OHQh0HoQ            The best thing to ever happen to iPhone      0   
27      jt2OHQh0HoQ  How to internet speed \nVideos \n👇👇👇👇👇👇👇👇👇👇\nh...      0   
28      jt2OHQh0HoQ                                  Sub to my channel      0   
29      jt2OHQh0HoQ                        24:57:34 which song is that      1   
...             ...                                                ...    ...   
718422  lLN1FwiqGwc  ...Was I the only one that liked the dancing s...      1   
718423  lLN1FwiqGwc                            MEME ME MEME ME MEME ME      0   
718424  lLN1FwiqGwc  Couldn't lure awkward fanart to Tumblr...........      0   
718425  lLN1FwiqGwc                            MEME ME MEME ME MEME ME      0   
718426  lLN1FwiqGwc                             Review happy death day      0   
718427  lLN1FwiqGwc                  Wa-Ha\nWa-Ha\nWa-Ha\nWa-Ha\nWa-Ha      0   
718428  lLN1FwiqGwc                         24:28 absolutely killed me      0   
718429  lLN1FwiqGwc                              #IT vs Freddy Krueger      0   
718430  lLN1FwiqGwc                                25:09 \nOh F*ck No!      0   
718431  lLN1FwiqGwc  Toll flot to!!!!!!!  Were. Freckling. Clown,s!...      0   
718432  lLN1FwiqGwc                         kudos for the mtg cards :D      0   
718433  lLN1FwiqGwc                     You have a fiancé Tamara...? 😰      0   
718434  lLN1FwiqGwc                               Damn good episode! 😆      0   
718435  lLN1FwiqGwc                                  WAHA! WAHA! WAHA!      0   
718436  lLN1FwiqGwc                                               Yes!      0   
718437  lLN1FwiqGwc                                          I love it      0   
718438  lLN1FwiqGwc     Bloody boogers...\n-Nostalgia Critic, IT (old)      0   
718439  lLN1FwiqGwc  MEME ME! MEME ME! MEM

## GB Videos
GB videos adalah dataset video di YouTube yang terdiri dari kolom sebagai berikut :
 - id video <br>
 - judul <br>
 - nama channel <br>
 - id kategori video <br>
 - tag <br>
 - jumlah penonton <br>
 - jumlah like <br>
 - jumlah dislike <br>
 - total komentar <br>
 - thumbnail <br>
 
Tetapi ada data yang tidak bisa dibaca sehingga harus diskip yaitu :
 - line 521402
 - line 2398
 - line 2797
 
Dimensi data ini adalah 7992 baris × 11 kolom

In [3]:
dfGBvideos

video_id                                              title  \
0     jt2OHQh0HoQ  Live Apple Event - Apple September Event 2017 ...   
1     AqokkXoa7uE  Holly and Phillip Meet Samantha the Sex Robot ...   
2     YPVcg45W0z4                    My DNA Test Results! I'm WHAT?!   
3     T_PuZBdT2iM  getting into a conversation in a language you ...   
4     NsjsmgmbCfc                               Baby Name Challenge!   
5     zZ2CLmvqfXg  REVEALED - FIFA 18 stats for Chelsea's Hazard,...   
6     w8fAellnPns  Juicy Chicken Breast - You Suck at Cooking (ep...   
7     LcZ2AuvxXNA                Nerf Bow Trick Shots | Dude Perfect   
8     lamseWekh8Y     Weekend in BRIGHTON with Mike! | Fleur De Vlog   
9     QBGaO89cBMI                                   Radiohead - Lift   
10    r4RfsfD6ZhM                           iPhone leaks (YIAY #355)   
11    gFIUXqcOulA                            Flinch w/ Fifth Harmony   
12    eM_FR7I2Ttw  Harry Styles - The Chain (Fleetwood Mac cover)...   
13    MdzGZv3zQ-U  Action Bronson and Sean Evans Have a Sandwich ...   
14    ql0Op1VcELw            What's Actually the Plane of the Future   
15    6vGg-jJl30A  THIS MADE MY DAD CRY! | Story of my Comeback C...   
16    -Ifnaxi2LQg          Fergie - You Already Know ft. Nicki Minaj   
17    TqG3syU5H7E                            WHY I STOPPED UPLOADING   
18    F2uJvwiSZAQ            How 9/11 changed Disney's Lilo & Stitch   
19    a7Sf_H2cFdM  Drowning for Power - Rooster Teeth Animated Ad...   
20    DuPXdnSWoLk               The Child in Time: Trailer - BBC One   
21    j6fn3LfIfww                                   Was It Worth It?   
22    LDcm6twPEJA                        The Problem With Our Phones   
23    hQbeB0YXLhs  Fenty Beauty Pro Filt'r Foundation Review|| Ny...   
24    yuwp7x_IiZ4  playing with old fridges -  public information...   
25    7wGDvthZ0ng  PewDiePie Getting TAKEN DOWN After Saying N Wo...   
26    VIlqTS96NFY  Autumn Hair, Lingerie & Lots More! August Favo...   
27    TeSoGKTZQl8  TESTING SOPH X MAKEUP REVOLUTION PALETTES!! | ...   
28    12fphklmFzk        Asuka is coming to Raw: Raw, Sept. 11, 2017   
29    GpAuCG6iUcA                   Fifty Shades Freed - Teaser [HD]   
...           ...                                                ...   
7963  M16CGK1T9MM  Why Photos of the Eiffel Tower at Night are Il...   
7964  E3DWd2dTiaQ                    Halloween Pumpkin Message Board   
7965  zA3VCYGG_OQ        SNIPER TOWER BUILD (Fortnite Battle Royale)   
7966  wm3aO5547Gc  GSP vs. Bisping: Threats, Cussing, Insults In ...   
7967  srWtQnseRyE                P!nk's Son Cries Whenever She Sings   
7968  L-zZFZiRsTk    MY 100 STEP SKINCARE ROUTINE FOR FLAWLESS SKIN!   
7969  2HcqS508lIo                The US Military's Most Powerful Gun   
7970  S_dN5PviiWI  Get Ready With Us After A Night Out - Melanie ...   
7971  7LfBoTAzjxM  Top Of Ultimate Retarded Drivers Fails, Extrem...   
7972  VBoFYQuBq_A                          VLOG : DANCE MAT QUEENS 👑   
7973  8I1B4n_8Cto                   Beck - Colors (Slime Visualizer)   
7974  0hLx_1Q15eY                      Where Do Our Drugs Come From?   
7975  AjioF-ZFDHQ  (WOW!) MALIGNAGGI REVEALS MCGREGOR FIGHT TALKS...   
7976  MAaOyRot_2s                           REUNITED AFTER 30 YEARS!   
7977  xNddRhpx5tA       Everything Wrong With Friday the 13th Part 2   
7978  x2IttXFBYxc      21 Savage - Nothin New (Official Music Video)   
7979  t8sgy0faXyg  Guns N' Roses Feat. PINK! - Patience (Msg, Nyc...   
7980  1wwwXCQyIRA                                FENTY BEAUTY REVIEW   
7981  EKH9TTHgVjA      ZARA, TOPSHOP, URBAN OUTFITTERS | Autumn Haul   
7982  DAbDvqPFXvQ  John Carpenter's Halloween by Trent Reznor & A...   
7983  LunHybOKIjU  Eminem Rips Donald Trump In BET Hip Hop Awards...   
7984  AUP3uhrM19c  Watford 2-1 Arsenal - Arsene Wenger Post Match...   
7985  xBU77C4cpTk      CARLIBYBELxMISSGUIDED FALL COLLECTION REVEAL!   
7986  8pHPohFezIk  Crystal Palace vs Chelsea 2-1 

## GB Category
GB category adalah dataset category video beserta idnya berbentuk json sehingga harus diolah ulang menjadi dataframe

In [4]:
jsGBcategory

{'kind': 'youtube#videoCategoryListResponse',
 'etag': '"m2yskBQFythfE4irbTIeOgYYfBU/1v2mrzYSYG6onNLt2qTj13hkQZk"',
 'items': [{'kind': 'youtube#videoCategory',
   'etag': '"m2yskBQFythfE4irbTIeOgYYfBU/Xy1mB4_yLrHy_BmKmPBggty2mZQ"',
   'id': '1',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdntcQ',
    'title': 'Film & Animation',
    'assignable': True}},
  {'kind': 'youtube#videoCategory',
   'etag': '"m2yskBQFythfE4irbTIeOgYYfBU/UZ1oLIIz2dxIhO45ZTFR3a3NyTA"',
   'id': '2',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdntcQ',
    'title': 'Autos & Vehicles',
    'assignable': True}},
  {'kind': 'youtube#videoCategory',
   'etag': '"m2yskBQFythfE4irbTIeOgYYfBU/nqRIq97-xe5XRZTxbknKFVe5Lmg"',
   'id': '10',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdntcQ',
    'title': 'Music',
    'assignable': True}},
  {'kind': 'youtube#videoCategory',
   'etag': '"m2yskBQFythfE4irbTIeOgYYfBU/HwXKamM1Q20q9BN-oBJavSGkfDI"',
   'id': '15',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdnt

### Mentransformasi json GB Category menjadi dataframe
Json GB category terdiri dari :
 - kind<br>
 - etag<br>
 - id<br>
 - items<br>
 - snippet<br>
 - title<br>
 - assignable<br>
 
items adalah list yang berisi kind, etag, id, snippet, title dan assignable.
yang dibutuhkan di sini adalah id kategori dan title / nama kategori. Maka 2 hal tersebut yang akan diekstrak kemudian dimasukkan ke dalam dataframe.

Dimensi data ini adalah 30 baris x 2 kolom

#### Kunci GB Category

In [5]:
jsGBcategory.keys()

dict_keys(['kind', 'etag', 'items'])

#### Normalisasi json GB Category
Karena yang dibutuhkan adalah id dan title di dalam list items, maka items harus dinormalisasi

In [6]:
jsGBcatitem = json_normalize(jsGBcategory['items'],sep=",")
jsGBcatitem

etag  id  \
0   "m2yskBQFythfE4irbTIeOgYYfBU/Xy1mB4_yLrHy_BmKm...   1   
1   "m2yskBQFythfE4irbTIeOgYYfBU/UZ1oLIIz2dxIhO45Z...   2   
2   "m2yskBQFythfE4irbTIeOgYYfBU/nqRIq97-xe5XRZTxb...  10   
3   "m2yskBQFythfE4irbTIeOgYYfBU/HwXKamM1Q20q9BN-o...  15   
4   "m2yskBQFythfE4irbTIeOgYYfBU/9GQMSRjrZdHeb1OEM...  17   
5   "m2yskBQFythfE4irbTIeOgYYfBU/FJwVpGCVZ1yiJrqZb...  18   
6   "m2yskBQFythfE4irbTIeOgYYfBU/M-3iD9dwK7YJCafRf...  19   
7   "m2yskBQFythfE4irbTIeOgYYfBU/WmA0qYEfjWsAoyJFS...  20   
8   "m2yskBQFythfE4irbTIeOgYYfBU/EapFaGYG7K0StIXVf...  21   
9   "m2yskBQFythfE4irbTIeOgYYfBU/xId8RX7vRN8rqkbYZ...  22   
10  "m2yskBQFythfE4irbTIeOgYYfBU/G9LHzQmx44rX2S5ya...  23   
11  "m2yskBQFythfE4irbTIeOgYYfBU/UVB9oxX2Bvqa_w_y3...  24   
12  "m2yskBQFythfE4irbTIeOgYYfBU/QiLK0ZIrFoORdk_g2...  25   
13  "m2yskBQFythfE4irbTIeOgYYfBU/r6Ck6Z0_L0rG37VJQ...  26   
14  "m2yskBQFythfE4irbTIeOgYYfBU/EoYkczo9I3RCf96Rv...  27   
15  "m2yskBQFythfE4irbTIeOgYYfBU/w5HjcTD82G_XA3xBc...  28   
16  "m2yskBQFythfE4irbTIeOgYYfBU/lL7uWDr_071CHxifj...  30   
17  "m2yskBQFythfE4irbTIeOgYYfBU/WnuVfjO-PyFLO7NTR...  31   
18  "m2yskBQFythfE4irbTIeOgYYfBU/ctpH2hGA_UZ3volJT...  32   
19  "m2yskBQFythfE4irbTIeOgYYfBU/L0kR3-g1BAo5UD1PL...  33   
20  "m2yskBQFythfE4irbTIeOgYYfBU/pUZOAC_s9sfiwar63...  34   
21  "m2yskBQFythfE4irbTIeOgYYfBU/Xb5JLhtyNRN3AQq02...  35   
22  "m2yskBQFythfE4irbTIeOgYYfBU/u8WXzF4HIhtEi805_...  36   
23  "m2yskBQFythfE4irbTIeOgYYfBU/D04PP4Gr7wc4IV_O9...  37   
24  "m2yskBQFythfE4irbTIeOgYYfBU/i5-_AceGXQCEEMWU0...  38   
25  "m2yskBQFythfE4irbTIeOgYYfBU/rtlxd0zOixA9QHdIZ...  39   
26  "m2yskBQFythfE4irbTIeOgYYfBU/N1TrDFLRppxZgBowC...  40   
27  "m2yskBQFythfE4irbTIeOgYYfBU/7UMGi6zRySqXopr_r...  41   
28  "m2yskBQFythfE4irbTIeOgYYfBU/RScXhi324h8usyIet...  42   
29  "m2yskBQFythfE4irbTIeOgYYfBU/0n9MJVCDLpA8q7aiG...  43   
30  "m2yskBQFythfE4irbTIeOgYYfBU/x5NxSf5fz8hn4loSN...  44   

                     kind  snippet,assignable         snippet,channelId  \
0   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
1   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
2   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
3   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
4   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
5   youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
6   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
7   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
8   youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
9   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
10  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
11  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
12  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
13  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
14  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
15  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
16  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
17  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
18  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
19  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
20  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
21  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
22  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
23  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
24  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
25  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
26  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ

#### Memasukkan isi dari items ke dalam dataframe dfGBcategory
items dimasukkan menjadi dataframe agar mudah untuk dikelola

In [7]:
dfGBcategory = jsGBcatitem
dfGBcategory

etag  id  \
0   "m2yskBQFythfE4irbTIeOgYYfBU/Xy1mB4_yLrHy_BmKm...   1   
1   "m2yskBQFythfE4irbTIeOgYYfBU/UZ1oLIIz2dxIhO45Z...   2   
2   "m2yskBQFythfE4irbTIeOgYYfBU/nqRIq97-xe5XRZTxb...  10   
3   "m2yskBQFythfE4irbTIeOgYYfBU/HwXKamM1Q20q9BN-o...  15   
4   "m2yskBQFythfE4irbTIeOgYYfBU/9GQMSRjrZdHeb1OEM...  17   
5   "m2yskBQFythfE4irbTIeOgYYfBU/FJwVpGCVZ1yiJrqZb...  18   
6   "m2yskBQFythfE4irbTIeOgYYfBU/M-3iD9dwK7YJCafRf...  19   
7   "m2yskBQFythfE4irbTIeOgYYfBU/WmA0qYEfjWsAoyJFS...  20   
8   "m2yskBQFythfE4irbTIeOgYYfBU/EapFaGYG7K0StIXVf...  21   
9   "m2yskBQFythfE4irbTIeOgYYfBU/xId8RX7vRN8rqkbYZ...  22   
10  "m2yskBQFythfE4irbTIeOgYYfBU/G9LHzQmx44rX2S5ya...  23   
11  "m2yskBQFythfE4irbTIeOgYYfBU/UVB9oxX2Bvqa_w_y3...  24   
12  "m2yskBQFythfE4irbTIeOgYYfBU/QiLK0ZIrFoORdk_g2...  25   
13  "m2yskBQFythfE4irbTIeOgYYfBU/r6Ck6Z0_L0rG37VJQ...  26   
14  "m2yskBQFythfE4irbTIeOgYYfBU/EoYkczo9I3RCf96Rv...  27   
15  "m2yskBQFythfE4irbTIeOgYYfBU/w5HjcTD82G_XA3xBc...  28   
16  "m2yskBQFythfE4irbTIeOgYYfBU/lL7uWDr_071CHxifj...  30   
17  "m2yskBQFythfE4irbTIeOgYYfBU/WnuVfjO-PyFLO7NTR...  31   
18  "m2yskBQFythfE4irbTIeOgYYfBU/ctpH2hGA_UZ3volJT...  32   
19  "m2yskBQFythfE4irbTIeOgYYfBU/L0kR3-g1BAo5UD1PL...  33   
20  "m2yskBQFythfE4irbTIeOgYYfBU/pUZOAC_s9sfiwar63...  34   
21  "m2yskBQFythfE4irbTIeOgYYfBU/Xb5JLhtyNRN3AQq02...  35   
22  "m2yskBQFythfE4irbTIeOgYYfBU/u8WXzF4HIhtEi805_...  36   
23  "m2yskBQFythfE4irbTIeOgYYfBU/D04PP4Gr7wc4IV_O9...  37   
24  "m2yskBQFythfE4irbTIeOgYYfBU/i5-_AceGXQCEEMWU0...  38   
25  "m2yskBQFythfE4irbTIeOgYYfBU/rtlxd0zOixA9QHdIZ...  39   
26  "m2yskBQFythfE4irbTIeOgYYfBU/N1TrDFLRppxZgBowC...  40   
27  "m2yskBQFythfE4irbTIeOgYYfBU/7UMGi6zRySqXopr_r...  41   
28  "m2yskBQFythfE4irbTIeOgYYfBU/RScXhi324h8usyIet...  42   
29  "m2yskBQFythfE4irbTIeOgYYfBU/0n9MJVCDLpA8q7aiG...  43   
30  "m2yskBQFythfE4irbTIeOgYYfBU/x5NxSf5fz8hn4loSN...  44   

                     kind  snippet,assignable         snippet,channelId  \
0   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
1   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
2   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
3   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
4   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
5   youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
6   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
7   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
8   youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
9   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
10  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
11  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
12  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
13  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
14  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
15  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
16  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
17  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
18  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
19  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
20  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
21  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
22  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
23  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
24  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
25  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
26  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ

### Mentransformasi dfGBcategory menjadi id dan nama category saja
id dan nama category/snippet title dibutuhkan untuk dimerge dengan dataframe GB comments dan GB videos

In [8]:
dfGBcategory1 = dfGBcategory.drop(['etag','kind','snippet,assignable','snippet,channelId'], axis=1)
dfGBcategory1

id         snippet,title
0    1      Film & Animation
1    2      Autos & Vehicles
2   10                 Music
3   15        Pets & Animals
4   17                Sports
5   18          Short Movies
6   19       Travel & Events
7   20                Gaming
8   21         Videoblogging
9   22        People & Blogs
10  23                Comedy
11  24         Entertainment
12  25       News & Politics
13  26         Howto & Style
14  27             Education
15  28  Science & Technology
16  30                Movies
17  31       Anime/Animation
18  32      Action/Adventure
19  33              Classics
20  34                Comedy
21  35           Documentary
22  36                 Drama
23  37                Family
24  38               Foreign
25  39                Horror
26  40        Sci-Fi/Fantasy
27  41              Thriller
28  42                Shorts
29  43                 Shows
30  44              Trailers

## Membaca csv datasets dari United States
Datasets Great Britain terdiri dari :
 - US Comments : datasets komentar dari tiap video
 - US Videos : datasets tentang video
 - US Category : datasets kategori video

In [9]:
#datasets US
dfUScomments = pd.read_csv('UScomments.csv', sep=',', skiprows=0, error_bad_lines=False)
dfUSvideos = pd.read_csv('USvideos.csv', sep=',', skiprows=0, error_bad_lines=False)

with open('GB_category_id.json') as file:    
    jsUScategory = js.load(file)

b'Skipping line 41589: expected 4 fields, saw 11\nSkipping line 51628: expected 4 fields, saw 7\nSkipping line 114465: expected 4 fields, saw 5\n'
b'Skipping line 142496: expected 4 fields, saw 8\nSkipping line 189732: expected 4 fields, saw 6\nSkipping line 245218: expected 4 fields, saw 7\n'
b'Skipping line 388430: expected 4 fields, saw 5\n'
C:\ProgramData\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3049: DtypeWarning: Columns (2,3) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
b'Skipping line 2401: expected 11 fields, saw 21\nSkipping line 2800: expected 11 fields, saw 21\nSkipping line 5297: expected 11 fields, saw 12\nSkipping line 5299: expected 11 fields, saw 12\nSkipping line 5300: expected 11 fields, saw 12\nSkipping line 5301: expected 11 fields, saw 12\n'


## US Comments
US comments adalah dataset komentar video YouTube di United States yang terdiri dari :
 - id video
 - teks komentar
 - berapa jumlah like dari komentar
 - berapa jumlah jawaban dari komentar
 
Tetapi ada data yang tidak dapat terbaca sehingga harus diskip yaitu :
 - line 41589
 - line 51628
 - line 114465
 - line 142496
 - line 189732 
 - line 245218
 - line 388430
 
Dimensi data ini adalah 691400 baris × 4 kolom

In [10]:
dfUScomments

video_id                                       comment_text likes  \
0       XpVt6Z1Gjjo                  Logan Paul it's yo big day ‼️‼️‼️     4   
1       XpVt6Z1Gjjo  I've been following you from the start of your...     3   
2       XpVt6Z1Gjjo                 Say hi to Kong and maverick for me     3   
3       XpVt6Z1Gjjo                                MY FAN . attendance     3   
4       XpVt6Z1Gjjo                                         trending 😉     3   
5       XpVt6Z1Gjjo                            #1 on trending AYYEEEEE     3   
6       XpVt6Z1Gjjo                               The end though 😭👍🏻❤️     4   
7       XpVt6Z1Gjjo                               #1 trending!!!!!!!!!     3   
8       XpVt6Z1Gjjo                        Happy one year vlogaversary     3   
9       XpVt6Z1Gjjo  You and your shit brother may have single hand...     0   
10      XpVt6Z1Gjjo             There should be a mini Logan Paul too!     0   
11      XpVt6Z1Gjjo  Dear Logan, I really wanna get your Merch but ...     0   
12      XpVt6Z1Gjjo  Honestly Evan is so annoying. Like its not fun...     0   
13      XpVt6Z1Gjjo                   Casey is still better then logan     0   
14      XpVt6Z1Gjjo      aw geez rick this guy is the face of YouTube.     0   
15      XpVt6Z1Gjjo                       He happy cause he in a movie     0   
16      XpVt6Z1Gjjo  Ayyyyoooo Logang what up .  This was a hard vl...     1   
17      XpVt6Z1Gjjo  Bro y didnt u give merch to johannes he is ur ...     0   
18      XpVt6Z1Gjjo  It's been fun watching you grow. I'm at 42 day...     3   
19      XpVt6Z1Gjjo             Made a lot of people hate youtube - GJ     0   
20      XpVt6Z1Gjjo  NO HONEY NOOO can't believe he has almost 12 m...     0   
21      XpVt6Z1Gjjo  Jake Paul is the fastest growing channel to hi...     1   
22      XpVt6Z1Gjjo  lol PEWDIEPIE accidentally played song with wo...     0   
23      XpVt6Z1Gjjo  You should do a prank that you broke  your pla...     0   
24      XpVt6Z1Gjjo  I love Logan and Jake so much and thay are so ...     1   
25      XpVt6Z1Gjjo                           #1 on Trending let's go.     0   
26      XpVt6Z1Gjjo            Evan was being such a dick on this vlog     0   
27      XpVt6Z1Gjjo  if you get allot of diss likes do you get on t...     0   
28      XpVt6Z1Gjjo  Evan is a horrible human being he also looks s...     0   
29      XpVt6Z1Gjjo  I love you so much I love to meet you But I li...     0   
...             ...                                                ...   ...   
691370  EoejGgUNmVU                                Suena genial-!!!!!!     0   
691371  EoejGgUNmVU                                 Maravilhosa !😘👏👏👏👏     0   
691372  EoejGgUNmVU                 🎶💁✌️❤️❤️❤️❤️❤️❤️❤️😔💕💕💕💕💕💕😍😍😍😻😻😻😻😻😻     0   
691373  EoejGgUNmVU                             Perfect. Every version     0   
691374  EoejGgUNmVU                                        Hammer Lied     0   
691375  EoejGgUNmVU                   shuni eshitgan Uzbeklar bormi...     0   
691376  EoejGgUNmVU                                  Потрясающий голос     0   
691377  EoejGgUNmVU                                     Sempre unica!💖     0   
691378  EoejGgUNmVU                                        Omg yeah 💓😍     0   
691379  EoejGgUNmVU                                            Wow...❤     0   
691380  EoejGgUNmVU                                 Simply AMAZING!❤❤❤     0   
691381  EoejGgUNmVU                                           HA Rİ KA     0   
691382  EoejGgUNmVU                                  fucking good ❤️❤️     0   
691383  EoejGgUNmVU                                  Me encantaaaaa <3     0   
691384  EoejGgUNmVU                                             Super💕     0   
691385  EoejGgUNmVU                                      Magnifique LP     0   
691386  EoejGgUNmVU                                                  💓     0   
691387  EoejGgUNmVU                                       Замечательно     0   
691388  Eoe

# US Videos
US videos adalah dataset video di YouTube yang terdiri dari kolom sebagai berikut :
 - id video <br>
 - judul <br>
 - nama channel <br>
 - id kategori video <br>
 - tag <br>
 - jumlah penonton <br>
 - jumlah like <br>
 - jumlah dislike <br>
 - total komentar <br>
 - thumbnail <br>
 
Tetapi ada data yang tidak bisa dibaca sehingga harus diskip yaitu :
 - line 2401
 - line 2800
 - line 5297
 - line 5299
 - line 5300
 - line 5301
 
Dimensi data ini adalah 7991 baris × 11 kolom

In [11]:
dfUSvideos

video_id                                              title  \
0     XpVt6Z1Gjjo  1 YEAR OF VLOGGING -- HOW LOGAN PAUL CHANGED Y...   
1     K4wEI5zhHB0            iPhone X — Introducing iPhone X — Apple   
2     cLdxuaxaQwc                                        My Response   
3     WYYvHb03Eog                          Apple iPhone X first look   
4     sjlHnJvXdQs                                  iPhone X (parody)   
5     cMKX2tE5Luk    The Disaster Artist | Official Trailer HD | A24   
6     8wNr-NQImFg       The Check In: HUD, Ben Carson and Hurricanes   
7     _HTXMhKWqnA                   iPhone X Impressions & Hands On!   
8     _ANP3HR1jsM                         ATTACKED BY A POLICE DOG!!   
9     zgLtEob6X-Q                 Honest Trailers - The Mummy (2017)   
10    Ayb_2qbZHm4                                Honest College Tour   
11    CsdzflTXBVQ  Best Floyd Mayweather Interview | Awkward Puppets   
12    l864IBj7cgw  Jennifer Lawrence Challenges Jimmy to an Axe T...   
13    4MkC65emkG4  Hand In Hand A Benefit For Hurricane Relief | MTV   
14    vu_9muoxT50  Colin Cloud: Mind Reader Predicts Your Tweets ...   
15    1L7JFN7tQLs    iPhone X Hands on - Everything you need to know   
16    ZQK1F0wz6z4                          What Do You Want to Eat?!   
17    T_PuZBdT2iM  getting into a conversation in a language you ...   
18    w8fAellnPns  Juicy Chicken Breast - You Suck at Cooking (ep...   
19    UCrBICYM0yM  Downsizing (2017) - Official Trailer - Paramou...   
20    -Ifnaxi2LQg          Fergie - You Already Know ft. Nicki Minaj   
21    B7YaMkCl3XA      Hurricane Irma death toll rises to 10 in Cuba   
22    5ywKal6-anc  Gigi Hadid Loses High Heel During Fashion Week...   
23    4Yue-q9Jdbk     SUPERFRUIT REACTS TO TEENS REACT TO SUPERFRUIT   
24    JhA1Wi9mrns         Kid Rock - Tennessee Mountain Top [Lyrics]   
25    EVp4-qjWVJE  Chargers vs. Broncos | NFL Week 1 Game Highlights   
26    LcZ2AuvxXNA                Nerf Bow Trick Shots | Dude Perfect   
27    MdzGZv3zQ-U  Action Bronson and Sean Evans Have a Sandwich ...   
28    1ZNZY-gd3K0  Joe Arpaio: Last Week Tonight with John Oliver...   
29    2wxyDrfwlXQ               What's In This 90s Mystery Caboodle?   
...           ...                                                ...   
7962  k9VmfRBkUuM                          Crane Removal - Timelapse   
7963  3y5A4paFOb4  Logan Paul - Outta My Hair [Official Music Video]   
7964  x4SheWJkn6s          Latinos React To The Selena Google Doodle   
7965  IOp8n5TzcSQ           Bailey Bryan - Too Good [Official Cover]   
7966  8Wb9KYnNxUc  Draymond Green: I don't know how serious the R...   
7967  adgAxnflvOE  EXCITED Dogs in Car Can't Stop Howling For The...   
7968  mPjgRKW_Jmk  Equifax: Last Week Tonight with John Oliver (HBO)   
7969  XPqko6Q0lpE                     John Oliver - Harvey Weinstein   
7970  AApylcoF9jA            Storm Ophelia makes landfall in Ireland   
7971  xTyTQgb-y-A  Jordan​ and​ ​Lindsay’s - Foxtrot - Dancing wi...   
7972  CEdAjI801Wo  How Jonathan Groff Got The Lead On Netflix's  ...   
7973  PeUgWK0pUIA            How Video Games are Turning into Movies   
7974  s8GrGvjhz0s  Blake Lively - Talks About Harvey Weinstein - GMA   
7975  RA9gg4qBb30                 John McCain Liberty Medal speech -   
7976  X2tejP45Lwc  28 Things I'd Tell My Younger Self! Life Lesso...   
7977  z2v_rL5C8C8  VINTAGE SHOPPING FOR THE FIRST TIME! (a mess lol)   
7978  -_Q5kO4YXFs  My baby hears me for the first time and is alm...   
7979  xGd8KIXqPJY  Katy Perry - Making Of “Swish Swish” Music Vid...   
7980  n530GFrdPY8                        Brad Paisley - Heaven South   
7981  3VSa-oARk-w  Monument Valley 2 - Available on Android & iOS...   
7982  Hlt3rA-oDao                                    Kellywise - SNL   
7983  Ni5Uc-DdMqk  Chris Hemsworth and Mark Ruffalo Australian In...   
7984  pfAKloMu4S0  What the Bannon vs. McConnell fight means for ...   
7985  CN4YL1KPDAA                           Pixel

## US Category
US category adalah dataset category video di United States beserta idnya berbentuk json sehingga harus diolah ulang menjadi dataframe

In [12]:
jsUScategory

{'kind': 'youtube#videoCategoryListResponse',
 'etag': '"m2yskBQFythfE4irbTIeOgYYfBU/1v2mrzYSYG6onNLt2qTj13hkQZk"',
 'items': [{'kind': 'youtube#videoCategory',
   'etag': '"m2yskBQFythfE4irbTIeOgYYfBU/Xy1mB4_yLrHy_BmKmPBggty2mZQ"',
   'id': '1',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdntcQ',
    'title': 'Film & Animation',
    'assignable': True}},
  {'kind': 'youtube#videoCategory',
   'etag': '"m2yskBQFythfE4irbTIeOgYYfBU/UZ1oLIIz2dxIhO45ZTFR3a3NyTA"',
   'id': '2',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdntcQ',
    'title': 'Autos & Vehicles',
    'assignable': True}},
  {'kind': 'youtube#videoCategory',
   'etag': '"m2yskBQFythfE4irbTIeOgYYfBU/nqRIq97-xe5XRZTxbknKFVe5Lmg"',
   'id': '10',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdntcQ',
    'title': 'Music',
    'assignable': True}},
  {'kind': 'youtube#videoCategory',
   'etag': '"m2yskBQFythfE4irbTIeOgYYfBU/HwXKamM1Q20q9BN-oBJavSGkfDI"',
   'id': '15',
   'snippet': {'channelId': 'UCBR8-60-B28hp2BmDPdnt

### Mentransformasi json US Category menjadi dataframe
Json US category terdiri dari :
 - kind<br>
 - etag<br>
 - id<br>
 - items<br>
 - snippet<br>
 - title<br>
 - assignable<br>
 
items adalah list yang berisi kind, etag, id, snippet, title dan assignable.
yang dibutuhkan di sini adalah id kategori dan title / nama kategori. Maka 2 hal tersebut yang akan diekstrak kemudian dimasukkan ke dalam dataframe.

Dimensi data ini adalah 30 baris x 2 kolom

#### Key US Category

In [13]:
jsUScategory.keys()

dict_keys(['kind', 'etag', 'items'])

#### Normalisasi json US Category
Karena yang dibutuhkan adalah id dan title di dalam list items, maka items harus dinormalisasi

In [14]:
jsUScatitem = json_normalize(jsGBcategory['items'],sep=",")
jsUScatitem

etag  id  \
0   "m2yskBQFythfE4irbTIeOgYYfBU/Xy1mB4_yLrHy_BmKm...   1   
1   "m2yskBQFythfE4irbTIeOgYYfBU/UZ1oLIIz2dxIhO45Z...   2   
2   "m2yskBQFythfE4irbTIeOgYYfBU/nqRIq97-xe5XRZTxb...  10   
3   "m2yskBQFythfE4irbTIeOgYYfBU/HwXKamM1Q20q9BN-o...  15   
4   "m2yskBQFythfE4irbTIeOgYYfBU/9GQMSRjrZdHeb1OEM...  17   
5   "m2yskBQFythfE4irbTIeOgYYfBU/FJwVpGCVZ1yiJrqZb...  18   
6   "m2yskBQFythfE4irbTIeOgYYfBU/M-3iD9dwK7YJCafRf...  19   
7   "m2yskBQFythfE4irbTIeOgYYfBU/WmA0qYEfjWsAoyJFS...  20   
8   "m2yskBQFythfE4irbTIeOgYYfBU/EapFaGYG7K0StIXVf...  21   
9   "m2yskBQFythfE4irbTIeOgYYfBU/xId8RX7vRN8rqkbYZ...  22   
10  "m2yskBQFythfE4irbTIeOgYYfBU/G9LHzQmx44rX2S5ya...  23   
11  "m2yskBQFythfE4irbTIeOgYYfBU/UVB9oxX2Bvqa_w_y3...  24   
12  "m2yskBQFythfE4irbTIeOgYYfBU/QiLK0ZIrFoORdk_g2...  25   
13  "m2yskBQFythfE4irbTIeOgYYfBU/r6Ck6Z0_L0rG37VJQ...  26   
14  "m2yskBQFythfE4irbTIeOgYYfBU/EoYkczo9I3RCf96Rv...  27   
15  "m2yskBQFythfE4irbTIeOgYYfBU/w5HjcTD82G_XA3xBc...  28   
16  "m2yskBQFythfE4irbTIeOgYYfBU/lL7uWDr_071CHxifj...  30   
17  "m2yskBQFythfE4irbTIeOgYYfBU/WnuVfjO-PyFLO7NTR...  31   
18  "m2yskBQFythfE4irbTIeOgYYfBU/ctpH2hGA_UZ3volJT...  32   
19  "m2yskBQFythfE4irbTIeOgYYfBU/L0kR3-g1BAo5UD1PL...  33   
20  "m2yskBQFythfE4irbTIeOgYYfBU/pUZOAC_s9sfiwar63...  34   
21  "m2yskBQFythfE4irbTIeOgYYfBU/Xb5JLhtyNRN3AQq02...  35   
22  "m2yskBQFythfE4irbTIeOgYYfBU/u8WXzF4HIhtEi805_...  36   
23  "m2yskBQFythfE4irbTIeOgYYfBU/D04PP4Gr7wc4IV_O9...  37   
24  "m2yskBQFythfE4irbTIeOgYYfBU/i5-_AceGXQCEEMWU0...  38   
25  "m2yskBQFythfE4irbTIeOgYYfBU/rtlxd0zOixA9QHdIZ...  39   
26  "m2yskBQFythfE4irbTIeOgYYfBU/N1TrDFLRppxZgBowC...  40   
27  "m2yskBQFythfE4irbTIeOgYYfBU/7UMGi6zRySqXopr_r...  41   
28  "m2yskBQFythfE4irbTIeOgYYfBU/RScXhi324h8usyIet...  42   
29  "m2yskBQFythfE4irbTIeOgYYfBU/0n9MJVCDLpA8q7aiG...  43   
30  "m2yskBQFythfE4irbTIeOgYYfBU/x5NxSf5fz8hn4loSN...  44   

                     kind  snippet,assignable         snippet,channelId  \
0   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
1   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
2   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
3   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
4   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
5   youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
6   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
7   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
8   youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
9   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
10  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
11  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
12  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
13  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
14  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
15  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
16  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
17  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
18  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
19  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
20  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
21  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
22  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
23  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
24  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
25  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
26  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ

#### Memasukkan isi dari items ke dalam dataframe dfGBcategory
items dimasukkan menjadi dataframe agar mudah untuk dikelola

In [15]:
dfUScategory = jsUScatitem
dfUScategory

etag  id  \
0   "m2yskBQFythfE4irbTIeOgYYfBU/Xy1mB4_yLrHy_BmKm...   1   
1   "m2yskBQFythfE4irbTIeOgYYfBU/UZ1oLIIz2dxIhO45Z...   2   
2   "m2yskBQFythfE4irbTIeOgYYfBU/nqRIq97-xe5XRZTxb...  10   
3   "m2yskBQFythfE4irbTIeOgYYfBU/HwXKamM1Q20q9BN-o...  15   
4   "m2yskBQFythfE4irbTIeOgYYfBU/9GQMSRjrZdHeb1OEM...  17   
5   "m2yskBQFythfE4irbTIeOgYYfBU/FJwVpGCVZ1yiJrqZb...  18   
6   "m2yskBQFythfE4irbTIeOgYYfBU/M-3iD9dwK7YJCafRf...  19   
7   "m2yskBQFythfE4irbTIeOgYYfBU/WmA0qYEfjWsAoyJFS...  20   
8   "m2yskBQFythfE4irbTIeOgYYfBU/EapFaGYG7K0StIXVf...  21   
9   "m2yskBQFythfE4irbTIeOgYYfBU/xId8RX7vRN8rqkbYZ...  22   
10  "m2yskBQFythfE4irbTIeOgYYfBU/G9LHzQmx44rX2S5ya...  23   
11  "m2yskBQFythfE4irbTIeOgYYfBU/UVB9oxX2Bvqa_w_y3...  24   
12  "m2yskBQFythfE4irbTIeOgYYfBU/QiLK0ZIrFoORdk_g2...  25   
13  "m2yskBQFythfE4irbTIeOgYYfBU/r6Ck6Z0_L0rG37VJQ...  26   
14  "m2yskBQFythfE4irbTIeOgYYfBU/EoYkczo9I3RCf96Rv...  27   
15  "m2yskBQFythfE4irbTIeOgYYfBU/w5HjcTD82G_XA3xBc...  28   
16  "m2yskBQFythfE4irbTIeOgYYfBU/lL7uWDr_071CHxifj...  30   
17  "m2yskBQFythfE4irbTIeOgYYfBU/WnuVfjO-PyFLO7NTR...  31   
18  "m2yskBQFythfE4irbTIeOgYYfBU/ctpH2hGA_UZ3volJT...  32   
19  "m2yskBQFythfE4irbTIeOgYYfBU/L0kR3-g1BAo5UD1PL...  33   
20  "m2yskBQFythfE4irbTIeOgYYfBU/pUZOAC_s9sfiwar63...  34   
21  "m2yskBQFythfE4irbTIeOgYYfBU/Xb5JLhtyNRN3AQq02...  35   
22  "m2yskBQFythfE4irbTIeOgYYfBU/u8WXzF4HIhtEi805_...  36   
23  "m2yskBQFythfE4irbTIeOgYYfBU/D04PP4Gr7wc4IV_O9...  37   
24  "m2yskBQFythfE4irbTIeOgYYfBU/i5-_AceGXQCEEMWU0...  38   
25  "m2yskBQFythfE4irbTIeOgYYfBU/rtlxd0zOixA9QHdIZ...  39   
26  "m2yskBQFythfE4irbTIeOgYYfBU/N1TrDFLRppxZgBowC...  40   
27  "m2yskBQFythfE4irbTIeOgYYfBU/7UMGi6zRySqXopr_r...  41   
28  "m2yskBQFythfE4irbTIeOgYYfBU/RScXhi324h8usyIet...  42   
29  "m2yskBQFythfE4irbTIeOgYYfBU/0n9MJVCDLpA8q7aiG...  43   
30  "m2yskBQFythfE4irbTIeOgYYfBU/x5NxSf5fz8hn4loSN...  44   

                     kind  snippet,assignable         snippet,channelId  \
0   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
1   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
2   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
3   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
4   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
5   youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
6   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
7   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
8   youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
9   youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
10  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
11  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
12  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
13  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
14  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
15  youtube#videoCategory                True  UCBR8-60-B28hp2BmDPdntcQ   
16  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
17  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
18  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
19  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
20  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
21  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
22  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
23  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
24  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
25  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ   
26  youtube#videoCategory               False  UCBR8-60-B28hp2BmDPdntcQ

### Mentransformasi dfUScategory menjadi id dan nama category saja
id dan nama category/snippet title dibutuhkan untuk dimerge dengan dataframe US comments dan US videos

In [16]:
dfUScategory1 = dfUScategory.drop(['etag','kind','snippet,assignable','snippet,channelId'], axis=1)
dfUScategory1

id         snippet,title
0    1      Film & Animation
1    2      Autos & Vehicles
2   10                 Music
3   15        Pets & Animals
4   17                Sports
5   18          Short Movies
6   19       Travel & Events
7   20                Gaming
8   21         Videoblogging
9   22        People & Blogs
10  23                Comedy
11  24         Entertainment
12  25       News & Politics
13  26         Howto & Style
14  27             Education
15  28  Science & Technology
16  30                Movies
17  31       Anime/Animation
18  32      Action/Adventure
19  33              Classics
20  34                Comedy
21  35           Documentary
22  36                 Drama
23  37                Family
24  38               Foreign
25  39                Horror
26  40        Sci-Fi/Fantasy
27  41              Thriller
28  42                Shorts
29  43                 Shows
30  44              Trailers